src/msg/async/ProtocolV2.cc

```diff
@@ -560,8 +561,12 @@ ssize_t ProtocolV2::write_message(Message *m, bool more) {
     ldout(cct, 1) << __func__ << " error sending " << m << ", "
                   << cpp_strerror(rc) << dendl;
   } else {
-    connection->logger->inc(
-        l_msgr_send_bytes, total_send_size - connection->outgoing_bl.length());
+    const auto sent_bytes = total_send_size - connection->outgoing_bl.length();
+    connection->logger->inc(l_msgr_send_bytes, sent_bytes);
+    if (crypto_counters) {
+      connection->logger->inc(crypto_counters->tx, sent_bytes);
+    }
+    ceph_assert(bool(session_stream_handlers.tx) == bool(crypto_counters));
```

Comment:
adding a note explaining the assert?

Reply:
Sure, will add. Basically secure-mode-specific counters make sense only if we're in secure mode.